nomos-blend/message/src/reward/mod.rs

```diff
@@ -8,132 +8,110 @@ pub use activity::ActivityProof;
 use nomos_core::sdp::SessionNumber;
 pub use session::SessionInfo;
 pub use token::BlendingToken;
-use tracing::warn;
 
-use crate::reward::session::SessionRandomness;
+pub use crate::reward::session::SessionRandomness;
 
 const LOG_TARGET: &str = "blend::message::reward";
 
-/// Collects blending tokens for the current session, while retaining those from
-/// the previous session which are used to produce an activity proof.
-pub struct BlendingTokenCollector {
-    current_session_tokens: BlendingTokensForSession,
-    current_session_randomness: SessionRandomness,
-    previous_session_tokens: Option<BlendingTokensForSession>,
+pub trait BlendingTokenCollector {
```

Comment:
Do we need to define a trait for this behaviour? We know what we are dealing with here. What is your opinion on this?